Sources/ApolloCodegenLib/ApolloSchemaOptions.swift

```diff
@@ -9,8 +9,48 @@ public struct ApolloSchemaOptions {
     case schemaDefinitionLanguage = "graphql"
   }
   
-  let apiKey: String?
-  let endpointURL: URL
+  /// How to attempt to download your schema
+  public enum DownloadMethod: Equatable {
+
+    case registry(_ settings: RegistrySettings)
+    ///   - endpointURL: The endpoint to hit to download your schema.
+    case introspection(endpointURL: URL)
+    
+    public struct RegistrySettings: Equatable {
```

Comment:
This is a separate object so that it can have new parameters added without breaking the enum.